# Lecture 05

**Date:** Sep 10, 2024

This lecture delves into the fundamental technique of sequence alignment, a cornerstone of comparative genomics and evolutionary studies. We'll explore how aligning DNA, RNA, or protein sequences unveils crucial insights into biological relationships and functions. The session will navigate through the nuances of global and local alignment strategies, illuminating their distinct applications in genomic analysis. We'll unpack the mathematical foundations of alignment algorithms, introducing dynamic programming as a powerful problem-solving approach. By examining the Needleman-Wunsch and Smith-Waterman algorithms, students will gain a deep understanding of how computational methods tackle the complex task of finding similarities between biological sequences. This comprehensive exploration aims to equip students with the theoretical knowledge and practical skills essential for interpreting genomic data in diverse biological contexts.

## Learning objectives

What you should be able to do after today's lecture:

1.  Define sequence alignment and explain its importance in bioinformatics and genomic analysis.
2.  Differentiate between global and local pairwise sequence alignment, providing examples of when each is most appropriate.
3.  Describe the basic principles of scoring systems in sequence alignment, including match, mismatch, and gap penalties.
4.  Explain the fundamental concept of dynamic programming as applied to sequence alignment algorithms.
5.  Outline the key steps of the Needleman-Wunsch algorithm for global alignment, using a simple example.
6.  Compare the Smith-Waterman algorithm for local alignment to the Needleman-Wunsch algorithm, highlighting their differences and specific applications.

## Readings

Relevant content for today's lecture.

TODO:

## Presentation

TODO:
