<h1 align="center">
<b>Lecture 04</b><br>
De novo genome assembly
</h1>
<p align="center">
<b>Date:</b> Sep 5, 2024
</p>

This lecture aims to provide a thorough understanding of de novo assembly techniques, their applications, and their crucial role in advancing genomic research.

## Learning objectives

What you should be able to do after today's lecture:

1.  Explain the fundamental challenge of reconstructing a complete genome.
2.  Describe and apply the principles of the greedy algorithm.
3.  Understand and construct de Bruijn graphs.

## Readings

Relevant content for today's lecture.

-   [Greedy algorithm](https://omics.crumblearn.org/genomics/assembly/de-novo/greedy/)
-   de Bruijn
    -   [John hopkins lecture](https://www.cs.jhu.edu/~langmea/resources/lecture_notes/assembly_dbg.pdf)
    -   [Why are de Bruijn graphs useful for genome assembly?](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531759/)
    -   [Read mapping on de Bruijn graphs](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1103-9)
    -   [Galaxy](https://training.galaxyproject.org/training-material/topics/assembly/tutorials/debruijn-graph-assembly/slides-plain.html)
    -   [Data science for HTS](https://data-science-sequencing.github.io/Win2018/lectures/lecture7/)

## Presentation

-   **Live link:** [slides.com/d/KOit8yE/live](https://slides.com/d/KOit8yE/live)
<!-- -   **Download:** [biosc1540-l04.pdf](/lectures/03/biosc1540-l04.pdf) -->

<iframe src="https://slides.com/aalexmmaldonado/biosc1540-l04/embed?byline=hidden&share=hidden" width="100%" height="600" title="BIOSC 1540: Lecture 04" scrolling="no" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
